usd/include/sdf/usd/World.hh

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef SDF_USD_WORLD_HH_
+#define SDF_USD_WORLD_HH_
+
+#include <string>
+
+#include <pxr/usd/usd/stage.h>
+
+#include "sdf/World.hh"
+#include "sdf/sdf_config.h"
+
+namespace usd
```

Comment:
Should this be namespaced under `sdf::<VERSION>`, like our other symbols in the library?

Reply:
it's weird but I'm getting undefined references ehn I added the version

Reply:
I tried adding the sdf version namespace, and found that to avoid undefined references, the namespace must be added in `World.cc` as well:
```diff
diff --git a/usd/src/World.cc b/usd/src/World.cc
index 226fb97c..e645c76f 100644
--- a/usd/src/World.cc
+++ b/usd/src/World.cc
@@ -34,6 +34,9 @@
 
 namespace usd
 {
+  // Inline bracke to help doxygen filtering.
+  inline namespace SDF_VERSION_NAMESPACE {
+  //
   sdf::Errors ParseSdfWorld(const sdf::World &_world, pxr::UsdStageRefPtr &_stage,
       const std::string &_path)
   {
@@ -63,4 +66,5 @@ namespace usd
 
     return errors;
   }
+  }
 }
```

I think the reason why the namespace needs to be added to the source file is because the component uses methods, not classes, so the way namespaces are used/linked in the source files for this component are different from the way usage/linking occurs for the core libsdformat files (I could be wrong though). Would it be worth adding the namespace to both the header and source files, or does the code for this component need to be refactored some way so that the namespace is only included in the header file?

Reply:
https://github.com/ignitionrobotics/sdformat/pull/817/commits/91629c07da112face018cf167d567dd2f6a86020

Reply:
@adlarkin I think the issue is you have to reference the same symbol in both the header and the source. If you put the function declaration inside a namespace in the header, you have two options in the source. (1) Write `sdf::Errors sdf::usd::ParseSdfWorld(...)`, (2) Put the the whole function definition inside the namespace. A lot of source files in libsdformat have `using namespace sdf` at the beginning. This ensures that when you reference a symbol, the `sdf` namespace is searched for that symbol. I guess this is a third option :)